src/reducers/viewconfig.ts

```diff
@@ -114,8 +124,46 @@ const normalizeTree = (tree: ComponentConfig[], startIndex: number = -1): ViewCo
     return flatTree;
 };
 
-export const getNodeById = (id: number, state: ViewConfigReducer): NormalizedComponentConfig | undefined =>
-    state.find(item => item.id === id);
+const composeNode = (item: NormalizedComponentConfig, state: ViewConfigReducer): ComponentConfig => {
```

Comment:
I kind of prefer 'denormalize'? It's not a perfect term but it does explain what it does right off the bat. (it's also what's being used already for the `facetconfig` reducer rn in your PR)
I think compose is a bit vague.